akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterSharding.scala

```diff
@@ -195,68 +246,112 @@ final class AdaptedClusterShardingImpl(system: ActorSystem[_]) extends ClusterSh
   private val cluster = Cluster(system)
   private val untypedSystem = system.toUntyped
   private val untypedSharding = akka.cluster.sharding.ClusterSharding(untypedSystem)
+  private val log: LoggingAdapter = Logging(untypedSystem, classOf[ClusterSharding])
 
   override def spawn[A](
-    behavior:           Behavior[A],
+    behavior:           String ⇒ Behavior[A],
     entityProps:        Props,
     typeKey:            EntityTypeKey[A],
     settings:           ClusterShardingSettings,
     maxNumberOfShards:  Int,
     handOffStopMessage: A): ActorRef[ShardingEnvelope[A]] = {
-    val extractor = new HashCodeMessageExtractor[A](maxNumberOfShards)
-    spawn(behavior, entityProps, typeKey, settings, extractor, defaultShardAllocationStrategy(settings), handOffStopMessage)
+    val extractor = new HashCodeMessageExtractor[A](maxNumberOfShards, handOffStopMessage)
+    spawn2(behavior, entityProps, typeKey, settings, extractor, defaultShardAllocationStrategy(settings))
   }
 
-  override def spawn[E, A](
-    behavior:           Behavior[A],
+  override def spawnJavadsl[A](
+    behavior:           EntityIdToBehavior[A],
     entityProps:        Props,
     typeKey:            EntityTypeKey[A],
     settings:           ClusterShardingSettings,
-    messageExtractor:   ShardingMessageExtractor[E, A],
-    handOffStopMessage: A): ActorRef[E] =
-    spawn(behavior, entityProps, typeKey, settings, messageExtractor, defaultShardAllocationStrategy(settings), handOffStopMessage)
+    maxNumberOfShards:  Int,
+    handOffStopMessage: A): ActorRef[ShardingEnvelope[A]] = {
+    val extractor = new HashCodeMessageExtractor[A](maxNumberOfShards, handOffStopMessage)
+    spawnJavadsl(behavior, entityProps, typeKey, settings, extractor, defaultShardAllocationStrategy(settings))
+  }
 
-  override def spawn[E, A](
-    behavior:           Behavior[A],
+  override def spawn3[E, A](
+    behavior:         String ⇒ Behavior[A],
+    entityProps:      Props,
+    typeKey:          EntityTypeKey[A],
+    settings:         ClusterShardingSettings,
+    messageExtractor: ShardingMessageExtractor[E, A]): ActorRef[E] =
+    spawn2(behavior, entityProps, typeKey, settings, messageExtractor, defaultShardAllocationStrategy(settings))
+
+  override def spawnJavadsl[E, A](
+    behavior:         EntityIdToBehavior[A],
+    entityProps:      Props,
+    typeKey:          EntityTypeKey[A],
+    settings:         ClusterShardingSettings,
+    messageExtractor: ShardingMessageExtractor[E, A]): ActorRef[E] =
+    spawnJavadsl(behavior, entityProps, typeKey, settings, messageExtractor, defaultShardAllocationStrategy(settings))
+
+  override def spawn2[E, A](
```

Comment:
to avoid the overload issue I think we can call this `spawnWithMessageExtractor`. It's more power user. And the 3rd variant (called spawn3) can be removed, i.e. only one power user api that includes all of the params